provider/kubernetes/kubernetes.go

```diff
@@ -1269,3 +1283,8 @@ func templateSafeString(value string) error {
 	_, err := strconv.Unquote(`"` + value + `"`)
 	return err
 }
+
+func isPem(data []byte) bool {
+	block, _ := pem.Decode(data)
```

Comment:
since according to RFC 7468  PEM blocks must begin with `-----BEGIN`, could you not first do a comparison to the first bytes in the slice to see if they contain a valid header, and fail fast without having to decode the whole dataset first?

Reply:
This would also allow a clean message to be logged : `Missing or malformed PEM header` sort of thing...

Reply:
@dtomcej Thanks for taking a look at this.

I think this should be sufficient in most cases, however, if the PEM is malformed (or malicious) it could still break configuration reconciliation. If you do not think this is worth the performance hit, I could modify the check to:

1. check the header/footer
2. validate the data is valid base64

I actually just realized that I should be looping over the  rest of the data, in the case of a multi document PEM file. So, if I do go that approach, I'll need to re-implement a good chunk of pem.Decode() anyway. 

wdyt?

Reply:
I think both would be reasonable. Fail fast on a malformed PEM header, then pem.Decode if the header is valid. A regex header validation would also work to break apart a multi-document PEM file for a looping decode.